File: CvManagementSystem/CvManagementSystem.Api/Contracts/Positions/CreateUpdateAccessRuleRequest.cs
using UserService.Api.Contracts.Attributes;
using UserService.Domain.Models;
using UserService.Domain.Models.Attributes;

namespace UserService.Api.Contracts.Positions;

public class CreateUpdateAccessRuleRequest
{
    public FilterOperator FilterOperator { get; set; }
    public CreateUpdateAttributeValueRequest AttributeValue { get; set; } = null!;
    public AttributeDataType AttributeDataType { get; set; }
}

File: Frontend/src/models/CreateUpdateAccessRuleRequest.ts
import {FilterOperator} from "./FilterOperator";
import {CreateUpdateAttributeValueRequest} from "./CreateUpdateAttributeValueRequest";
import {AttributeDataType} from "./AttributeDataType";

export interface CreateUpdateAccessRuleRequest {
    filterOperator: FilterOperator;
    attributeValue: CreateUpdateAttributeValueRequest;
    attributeDataType: AttributeDataType;
    attributeValueId: string;
}